ido/src/modules/project.rs

```diff
@@ -352,17 +354,157 @@ impl IDOContract {
+            let account_id = env::signer_account_id();
+            let project = self.internal_get_project_or_panic(project_id);                         
+            let deposit = env::attached_deposit();
+
+            project.assert_sale_period();
+            assert!(self.is_whitelist(project_id,None),"Account do not register whitelisting this project");
+              
+            match project.sale_type {
+                SaleType::Shared { 
+                    min_allocation_per_user, 
+                    max_allocation_per_user} => {
+                        self.internal_sale_commit_shared_project(min_allocation_per_user, 
+                                                                max_allocation_per_user,
+                                                                project_id,
+                                                                account_id,
+                                                                deposit);
+                    },
+                SaleType::Lottery { 
+                    allocation_per_ticket, 
+                    total_tickets: _, 
+                    win_ticket_ids: _ } => {
+                        self.internal_sale_commit_lottery_project(allocation_per_ticket,
+                                                                project_id,
+                                                                account_id,
+                                                                deposit);
+                    }
+            }
+        }
+
+        #[payable]
+        pub fn internal_sale_commit_shared_project(&mut self,
+            min_allocation: u128, 
+            max_allocation: u128,
+            project_id: ProjectId, 
+            account_id: AccountId, 
+            deposit: u128) {
+                // Update Project struct
+                let mut project_account_unordered_map = self.internal_get_accounts_by_project_or_panic(project_id);
+                let mut project_account = self.internal_get_account_by_project_or_panic(project_id,&account_id);
+                let mut project = self.internal_get_project_or_panic(project_id);
+                let account_sale = project_account.sale_data.unwrap_or(
+                    AccountSale{
+                        committed_amount: 0,
+                        sale_data: AccountSaleData::Shared
+                    }
+                );
+                
+                assert!( (account_sale.committed_amount + deposit) > min_allocation 
+                    && (account_sale.committed_amount + deposit) < max_allocation,
+                    "Total deposit amount must be between min_allocation and max_allocation");
+
+                let account_sale = AccountSale{
+                    committed_amount: account_sale.committed_amount + deposit,
+                    sale_data: AccountSaleData::Shared
+                };
+                
+                // Update Project account
+                project_account = ProjectAccount{
```

Comment:
We just need to directly update `project_account.sale_data` and keep `project_account.distribution_data` as whatever it is. So it should be `project_account.sale_data = Some(account_sale)`.